java/core/page-factory-utils/src/main/java/ru/sbtqa/tag/pagefactory/util/PageFactoryUtils.java

```diff
@@ -231,5 +230,37 @@ public static void fireValidationRule(Page page, String title, Object... params)
         }
         throw new PageException("There is no '" + title + "' validation rule in '" + page.getPageTitle() + "' page.");
     }
-    
+
+    /**
+     * Search for the given element among the parent object fields,
+     * check whether it has a {@link
+     * RedirectsTo} annotation, and return a redirection page class, if so.
+     * Search goes in recursion if it meets HtmlElement field, as given
+     * element could be inside of the block
+     *
+     * @param element element that is being checked for redirection
+     * @param parent parent object
+     * @return class of the page, this element redirects to
+     */
+    public static Class<? extends Page> findRedirect(Object parent, Object element) {
+        List<Field> fields = FieldUtilsExt.getDeclaredFieldsWithInheritance(parent.getClass());
+
+        for (Field field : fields) {
+            RedirectsTo redirect = field.getAnnotation(RedirectsTo.class);
+            if (redirect != null) {
+                try {
+                    field.setAccessible(true);
+                    Object targetField = field.get(parent);
+                    if (targetField != null) {
+                        if (targetField == element) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)